src/Raven.Client/Documents/Operations/TimeSeries/TimeSeriesCollectionConfiguration.cs

```diff
@@ -0,0 +1,229 @@
+                return RawPolicy;
+
+            return Policies.SingleOrDefault(p => name.IndexOf(p.Name, StringComparison.InvariantCultureIgnoreCase) > 0);
+        }
+
+        internal TimeSeriesPolicy GetNextPolicy(TimeSeriesPolicy policy)
+        {
+            if (Policies.Count == 0)
+                return null;
+
+            if (policy == RawPolicy)
+                return Policies[0];
+
+            var current = Policies.FindIndex(p => p == policy);
+            if (current < 0)
+            {
+                Debug.Assert(false,"shouldn't happened, this mean the current policy doesn't exists");
+                return null;
+            }
+
+            if (current == Policies.Count - 1)
+                return TimeSeriesPolicy.AfterAllPolices;
+
+            return Policies[current + 1];
+        }
+
+        internal TimeSeriesPolicy GetPreviousPolicy(TimeSeriesPolicy policy)
+        {
+            if (policy == RawPolicy)
+                return TimeSeriesPolicy.BeforeAllPolices;
+
+            var current = Policies.FindIndex(p => p == policy);
+            if (current < 0)
+                return null;
+
+            if (current == 0)
+                return RawPolicy;
+
+            return Policies[current - 1];
+        }
+
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(Policies)] = new DynamicJsonArray(Policies.Select(p=>p.ToJson())),
+                [nameof(RawPolicy)] = RawPolicy.ToJson(),
+                [nameof(Disabled)] = Disabled
+            };
+        }
+    }
+
+    public class RawTimeSeriesPolicy : TimeSeriesPolicy
+    {
+        internal const string PolicyString = "rawpolicy"; // must be lower case
+        public RawTimeSeriesPolicy()
+        {
+            // for de-serializer
+        }
+
+        public RawTimeSeriesPolicy(TimeSpan? retentionTime)
+        {
+            if (retentionTime <= TimeSpan.Zero)
+                throw new ArgumentException("Must be greater than zero", nameof(retentionTime));
+
+            Name = PolicyString;
+            RetentionTime = retentionTime;
+        }
+    }
+
+    public class TimeSeriesPolicy : IDynamicJson, IComparable<TimeSeriesPolicy>
+    {
+        /// <summary>
+        /// Name of the time series policy, defined by convention
+        /// </summary>
+        public string Name { get; protected set; }
+
+        /// <summary>
+        /// How long the data of this policy will be retained
+        /// </summary>
+        public TimeSpan? RetentionTime { get; protected set; }
+
+        /// <summary>
+        /// Define the aggregation of this policy
+        /// </summary>
+        public TimeSpan AggregationTime { get; private set; }
+
+        /// <summary>
+        /// Define the aggregation type
+        /// </summary>
+        public AggregationType Type { get; private set; }
```

Comment:
This actually an interesting issue.
Let's say we have a single value time-series. Stock price. 

Right now each aggregation result is represented by a point on the policy time-series. So if I aggregate daily by `Average`, each day I'll create a single point from the raw stock price.

But if I want to aggregate daily by `First`, `Last`, `High`, `Low`.
Right now it is a problem, since I can define only a single type.
To allow this kind of thing we can either create a complex policy tree or create a single 4-dim time-series (which I like more).

But do we want to tackle this now?

Reply:
The raw time series will be always aggregated into (`First`,`Last`,`Min`,`Max`,`Sum`,`Count`).
In that order.
Other policies will aggregate each of those values further independently.